src/SeqCli/Forwarder/Channel/ForwardingChannelMap.cs

```diff
@@ -45,30 +48,65 @@ ForwardingChannel OpenOrCreateChannel(string? apiKey, string name)
             apiKey,
             _shutdownTokenSource.Token);
     }
-
-    public ForwardingChannel Get(string? apiKey)
+    
+    void ReopenApiKeyChannels()
     {
-        if (string.IsNullOrWhiteSpace(apiKey))
+        if (_config.Forwarder.UseApiKeyForwarding)
         {
-            return _defaultChannel;
+            foreach (var directoryPath in Directory.EnumerateDirectories(_bufferPath))
+            {
+                if (directoryPath.Equals(GetStorePath("Default"))) continue;
+                
+                var path = new SystemStoreDirectory(directoryPath);
+                var apiKey = path.ReadApiKey(_config);
+
+                if (!string.IsNullOrEmpty(apiKey))
+                {
+                    var created = OpenOrCreateChannel(apiKey, ApiKeyToName(apiKey));
+                    
+                    lock (_channelsSync)
```

Comment:
Since this is only called in the constructor where there's no parallelism, it might be best to omit the `lock` statement here so the code doesn't give the impression of there being some.

Renaming the method to `LoadApiKeyChannels` or `InitApiKeyChannels` or `OpenApiKeyChannels` might communicate its role better?